test/models/test_rbm.py

```diff
@@ -59,16 +59,30 @@ def test_RBMSymm(use_hidden_bias, use_visible_bias, symmetries):
 @pytest.mark.parametrize("use_bias", [True, False])
 @pytest.mark.parametrize("symmetries", ["trans", "autom"])
 @pytest.mark.parametrize("lattice", [nk.graph.Chain, nk.graph.Square])
-def test_gcnn(use_bias, symmetries, lattice):
+@pytest.mark.parametrize("mode", ["fft", "irreps"])
+def test_gcnn(use_bias, symmetries, lattice, mode):
     g, hi, perms = _setup_symm(symmetries, N=3, lattice=lattice)
 
-    ma = nk.models.GCNN(
-        symmetries=perms,
-        layers=4,
-        features=4,
-        use_bias=use_bias,
-        bias_init=nk.nn.initializers.uniform(),
-    )
+    if mode == "fft":
+        ma = nk.models.GCNN(
+            symmetries=perms,
+            mode=mode,
+            shape=g.extent,
+            layers=2,
+            features=2,
+            use_bias=use_bias,
+            bias_init=nk.nn.initializers.uniform(),
+        )
+    else:
+        ma = nk.models.GCNN(
+            symmetries=perms,
+            mode=mode,
+            layers=2,
+            features=2,
+            use_bias=use_bias,
+            bias_init=nk.nn.initializers.uniform(),
+        )
+
```

Comment:
```suggestion
```

Reply:
The point of this is that when mode="irreps" you can't feed the kwarg "shape" or it will complain. There is probably a way I could condense this code, which I'll look into.

Reply:
ah! i missed that. sorry.
Why will it complain?
I think extra kwargs should not raise errors.

Reply:
I've seen it happen with Flax modules

Reply:
I get the following error message:

TypeError: __init__() got an unexpected keyword argument 

Is there a way to have flax modules take kwargs?

Reply:
How is that possible?
GCNN is a constructor function that you defined. 
One argument is shape.
If your constructor does the right thing, he will not be passing `shape` down to things that don't need it.
That's the whole point of having a constructor.

Reply:
I just tried by cloning your branch.

This whole set of if mode == elif elif can be substituted by a simple
```
    ma = nk.models.GCNN(
        symmetries=perms,
        mode=mode,
        shape=tuple(g.extent),
        layers=2,
        features=2,
        use_bias=use_bias,
        bias_init=nk.nn.initializers.uniform(),
    )
```

which will always work.
I don't see any problem.

Reply:
This does work! I'm now remembering that I fixed the constructor function.